### PR TITLE
finetune : zero the loraB initial vectors

### PR DESCRIPTION
**Pull request by AndrewGodfrey**
The LoRA paper recommends zeroing loraB, but the current code initializes both loraB and loraA to a normal distribution. In effect, we are randomizing the base model somewhat, before we start finetuning. (Because loraA*loraB - 'delta-W' in the paper - is starting at non-zero).

The numbers are small and in my own testing I couldn't see much difference. But then I saw the discussion on #3852. That bug says it's specific to Mistral/Zephyr, so maybe something else is a factor there, but I see they're using much higher values for r than I was, and in some cases also using --lora-alpha > --lora-r. So, I'm curious how much this PR affects that problem.

P.S. I also looked at loralib. It, too, zeroes at least one of the init vector pairs. (Though it departs from the paper in using a different distribution for loraA, in some cases).

### CODE REVIEW

**Comment by AndrewGodfrey** [comment]
Also, #3852 is looking specifically at model perplexity after a single training iteration. My testing was using 30 iterations. Perhaps there's some other training bug (because the perplexity after 30 iterations isn't great in either case). And I did notice a difference during the first few iterations in the 'loss' value:

<details><summary>before:</summary>

```	train_opt_callback: iter=     0 sample=1/26766 sched=0.000000 loss=0.000000 |->
	train_opt_callback: iter=     1 sample=5/26766 sched=0.010000 loss=4.155559 dt=00:03:08 eta=01:31:10 |->
	train_opt_callback: iter=     2 sample=9/26766 sched=0.020000 loss=4.215462 dt=00:03:02 eta=01:25:04 |>
	train_opt_callback: iter=     3 sample=13/26766 sched=0.030000 loss=4.099679 dt=00:03:10 eta=01:25:47 |-->
	train_opt_callback: iter=     4 sample=17/26766 sched=0.040000 loss=3.806161 dt=00:03:09 eta=01:22:11 |---->
	train_opt_callback: iter=     5 sample=21/26766 sched=0.050000 loss=3.946302 dt=00:02:45 eta=01:09:03 |--->
	train_opt_callback: iter=     6 sample=25/26766 sched=0.060000 loss=3.931692 dt=00:02:35 eta=01:02:15 |--->
	train_opt_callback: iter=     7 sample=29/26766 sched=0.070000 loss=4.179832 dt=00:02:36 eta=00:59:53 |->
	train_opt_callback: iter=     8 sample=33/26766 sched=0.080000 loss=3.153435 dt=00:02:36 eta=00:57:24 |----------->
	train_opt_callback: iter=     9 sample=37/26766 sched=0.090000 loss=3.967691 dt=00:02:32 eta=00:53:30 |--->
```
</details>
<details><summary>after:</summary>

```	train_opt_callback: iter=     0 sample=1/26766 sched=0.000000 loss=0.000000 |->
	train_opt_callback: iter=     1 sample=5/26766 sched=0.010000 loss=3.622779 dt=00:02:59 eta=01:26:52 |->
	train_opt_callback: iter=     2 sample=9/26766 sched=0.020000 loss=3.852748 dt=00:02:42 eta=01:16:01 |>
	train_opt_callback: iter=     3 sample=13/26766 sched=0.030000 loss=3.899308 dt=00:02:38 eta=01:11:24 |>
	train_opt_callback: iter=     4 sample=17/26766 sched=0.040000 loss=3.993236 dt=00:02:35 eta=01:07:17 |>
	train_opt_callback: iter=     5 sample=21/26766 sched=0.050000 loss=3.399619 dt=00:02:33 eta=01:03:59 |--->
	train_opt_callback: iter=     6 sample=25/26766 sched=0.060000 loss=3.960002 dt=00:02:30 eta=01:00:22 |>
	train_opt_callback: iter=     7 sample=29/26766 sched=0.070000 loss=3.472059 dt=00:02:32 eta=00:58:28 |--->
	train_opt_callback: iter=     8 sample=33/26766 sched=0.080000 loss=3.841346 dt=00:02:33 eta=00:56:13 |>
	train_opt_callback: iter=     9 sample=37/26766 sched=0.090000 loss=3.931076 dt=00:02:36 eta=00:54:55 |>
```

**Comment by AndrewGodfrey** [comment]
@xaedes fyi

File: common/train.cpp
Comment (gwjr) [inline common/train.cpp:1]: Is this intended to do anything different to [`ggml_set_zero()`](https://github.com/ggerganov/llama.cpp/blob/6bb4908a17150b49373b5f977685b2e180a04f6f/ggml.h#L709)?

File: common/train.cpp [hunk 1]
```diff
@@ -68,6 +68,51 @@ void free_random_uniform_distribution(struct random_uniform_distribution * rnd)
     free(rnd);
 }
 
+struct ggml_tensor * zero_tensor(struct ggml_tensor * tensor) {
```
Comment (Galunid) [inline common/train.cpp:71]: This *seems* to be the same as `ggml_set_zero`, but way less efficient

File: common/train.cpp
Comment (AndrewGodfrey) [reply common/train.cpp:1]: Thanks! I had looked for a function like that, but hadn't found it. I'll use that instead.

**Review by Galunid — APPROVED** [review]
Can we merge?
